src/bin/rad-ci.rs

```diff
@@ -0,0 +1,161 @@
+use std::convert::TryInto;
+use std::io::{Read, Write};
+use std::net::TcpStream;
+
+use rad_ci::{Options, HELP};
+use rad_terminal::components as term;
+
+use anyhow::Context;
+use ssh2::Session;
+
+fn main() {
+    rad_terminal::args::run_command::<Options, _>(HELP, "CI", run);
+}
+
+fn run(options: Options) -> anyhow::Result<()> {
+    let seed = options.seed.context("Seed is invalid")?;
+    // SSH into server
+    let tcp = TcpStream::connect(format!("{}:22", &seed))?;
+    let mut sess = Session::new()?;
+    sess.set_tcp_stream(tcp);
+    sess.handshake()?;
+
+    // Authenticate
+    let spinner = term::spinner(&format!("SSH into remote {}", &seed));
+    if options.ssh_index == 0 {
+        sess.userauth_agent(&options.ssh_user)?;
+    } else {
+        let mut agent = sess.agent()?;
+        agent.connect()?;
+        agent.list_identities()?;
+        let identities = agent.identities()?;
+        let identity = identities.get(options.ssh_index).context(format!(
+            "Couldn't retrieve ssh-agent key #{}",
+            options.ssh_index
+        ))?;
+        agent.userauth(&options.ssh_user, identity)?;
+    }
+    if sess.authenticated() {
+        spinner.finish();
+    } else {
+        spinner.failed();
```

Comment:
No point in continuing if we couldn't auth.

Reply:
It doesn't reach the `else`, either gets:

```
== CI failed [Session(-34)] no identities found in the ssh agent
```
or
```
== CI failed [Session(-18)] Username/PublicKey combination invalid
```